src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  let copyOfState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    if (actions[i].type === 'addProperties') {
+      Object.assign(copyOfState, actions[i].extraData);
+    }
+
+    if (actions[i].type === 'removeProperties') {
+      actions[i].keysToRemove.forEach((el) => delete copyOfState[el]);
+    }
+
+    if (actions[i].type === 'clear') {
+      Object.keys(copyOfState).forEach((key) => delete copyOfState[key]);
+    }
+
+    states.push(copyOfState);
+    copyOfState = { ...copyOfState };
```

Comment:
The current approach of pushing `copyOfState` into the `states` array and then cloning it afterwards is problematic. Since objects in JavaScript are reference types, all entries in the `states` array will end up pointing to the same object. To fix this, you should clone `copyOfState` before pushing it into the `states` array.